dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
+    file_path, file_row_group_stats, file_row_group_column_stats, stat_col_indices
+):
+    """ Utility to aggregate the statistics for N row-groups
+        into a single dictionary.
+    """
+    if len(file_row_group_stats) < 1:
+        raise ValueError("Empty statistics.")
+    elif len(file_row_group_column_stats) == 0:
+        assert len(file_row_group_stats) == 1
+        return file_row_group_stats[0]
+    else:
+        df_rgs = pd.DataFrame(file_row_group_stats)
+        s = {
+            "file_path_0": file_path,
+            "num-rows": df_rgs["num-rows"].sum(),
+            "total_byte_size": df_rgs["total_byte_size"].sum(),
+            "columns": [],
+        }
+        df_cols = pd.DataFrame(file_row_group_column_stats)
+        for ind, name in enumerate(stat_col_indices):
+            i = ind * 3
+            s["columns"].append(
+                {
+                    "name": name,
+                    "min": df_cols.iloc[:, i].min(),
+                    "max": df_cols.iloc[:, i + 1].max(),
+                    "null_count": df_cols.iloc[:, i + 2].sum(),
+                }
             )
+        return s
 
-        column_names, index_names = _normalize_index_columns(
-            columns, column_names + partitions, index, index_names
-        )
 
-        all_columns = index_names + column_names
+def _process_metadata(
+    metadata, single_rg_parts, gather_statistics, stat_col_indices, no_filters
+):
+    # Get the number of row groups per file
+    file_row_groups = defaultdict(list)
+    file_row_group_stats = defaultdict(list)
+    file_row_group_column_stats = defaultdict(list)
+    cmax_last = {}
+    for rg in range(metadata.num_row_groups):
+        row_group = metadata.row_group(rg)
+        fpath = row_group.column(0).file_path
+        if fpath is None:
+            raise ValueError("metadata is missing file_path string.")
+        if file_row_groups[fpath]:
+            file_row_groups[fpath].append(file_row_groups[fpath][-1] + 1)
+        else:
+            file_row_groups[fpath].append(0)
+        if gather_statistics:
+            if single_rg_parts:
+                s = {
+                    "file_path_0": fpath,
+                    "num-rows": row_group.num_rows,
+                    "total_byte_size": row_group.total_byte_size,
+                    "columns": [],
+                }
+            else:
+                s = {
+                    "num-rows": row_group.num_rows,
+                    "total_byte_size": row_group.total_byte_size,
+                }
+            cstats = []
+            for name, i in stat_col_indices.items():
+                column = row_group.column(i)
+                if column.statistics:
+                    cmin = column.statistics.min
+                    cmax = column.statistics.max
+                    cnull = column.statistics.null_count
+                    last = cmax_last.get(name, None)
+                    if no_filters and last and cmin < last:
```

Comment:
Oh, you do it here; I suppose you only pass on stats to the function above if you get something useful here?

Reply:
There might be an edge case to consider if all the values of the column happen to be null (we don't actually use `cnull` otherwise, right?)

Reply:
Hmm -  Good point

Reply:
I actually don't know what the right divisions to set would be for this case, if the column is an index. I *think* to keep repeating the same value.

@mrocklin , @TomAugspurger : what is the right thing to do with divisions for a partition where all the index values are null? Interestingly, the following reorders the rows and gathers the null index ones into a partition with divisions (None, None), which I suspect is not intentional
```
>>> df = dd.from_pandas(pd.DataFrame({'a': [0] * 10}, index=[1, None, None, None, None, 4, 5, 8, 9, 10]), npartitions=5)
>>> df.divisions
(1, 5, 9, None, None)
>>> df.compute()
     a
1    0
4    0
5    0
8    0
9    0
10   0
NaN  0
NaN  0
NaN  0
NaN  0
```

Reply:
It probably isn't well defined. Though I think that this is enough of an edge case to not worry too much about. NA values in indexes are broken in many places in pandas.

Reply:
OK, thanks.
So we can repeat the min value from the previous max value, or lists Nones, so long as when filtering happens, we don't end up loading the empty partition by accident (wouldn't be the end of the world, but should be able to determine for this case). I don't think we can current filter specifically for null values anyway.

Reply:
Okay - So it seems that having partitions with all-null values in the index is somewhat of an "undefined" case - However, the code probably shouldn't crash.  For this PR, does it make sense to bail on stats gathering (divisions) when there is an all-null partition?  This seems to be what fastparquet is doing.

Reply:
Bailing makes sense to me.